kms/uri/uri.go

```diff
@@ -0,0 +1,86 @@
+package uri
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// URI implements a parser for a URI format based on the the PKCS #11 URI Scheme
+// defined in https://tools.ietf.org/html/rfc7512
+//
+// These URIs will be used to define the key names in a KMS.
+type URI struct {
+	*url.URL
+	Values url.Values
```

Comment:
Why do we keep the values separate when they're also part of the opaque part of the URL?

Reply:
Because they are a raw-opaque string in the url, and they will have to be decoded every time we want to get one field, they are not part of the url.Query.